torchmetrics/regression/tweedie_deviance.py

```diff
@@ -0,0 +1,120 @@
+        \end{cases}
+
+    where :math:`y` is a tensor of targets values, and :math:`\hat{y}` is a tensor of predictions.
+
+    Forward accepts
+
+    - ``preds`` (float tensor): ``(N,)``
+    - ``targets`` (float tensor): ``(N,)``
+
+    Args:
+        power:
+            - power < 0 : Extreme stable distribution. (Requires: preds > 0.)
+            - power = 0 : Normal distribution. (Requires: targets and preds can be any real numbers.)
+            - power = 1 : Poisson distribution. (Requires: targets >= 0 and y_pred > 0.)
+            - 1 < p < 2 : Compound Poisson distribution. (Requires: targets >= 0 and preds > 0.)
+            - power = 2 : Gamma distribution. (Requires: targets > 0 and preds > 0.)
+            - power = 3 : Inverse Gaussian distribution. (Requires: targets > 0 and preds > 0.)
+            - otherwise : Positive stable distribution. (Requires: targets > 0 and preds > 0.)
+        compute_on_step:
+            Forward only calls ``update()`` and return ``None`` if this is set to ``False``.
+        dist_sync_on_step:
+            Synchronize metric state across processes at each ``forward()``
+            before returning the value at the step.
+        process_group:
+            Specify the process group on which synchronization is called.
+            default: ``None`` (which selects the entire world)
+        dist_sync_fn:
+            Callback that performs the allgather operation on the metric state. When ``None``, DDP
+            will be used to perform the all gather.
+
+    Example:
+        >>> from torchmetrics import TweedieDevianceScore
+        >>> targets = torch.tensor([1.0, 2.0, 3.0, 4.0])
+        >>> preds = torch.tensor([4.0, 3.0, 2.0, 1.0])
+        >>> deviance_score = TweedieDevianceScore(power=0)
+        >>> deviance_score(preds, targets)
+        tensor(5.)
+
+    """
+    sum_deviance_score: Tensor
+    num_observations: Tensor
+
+    def __init__(
+        self,
+        power: float = 0.0,
+        compute_on_step: bool = True,
+        dist_sync_on_step: bool = False,
+        process_group: Optional[Any] = None,
+        dist_sync_fn: Callable = None,
+    ) -> None:
+        super().__init__(
+            compute_on_step=compute_on_step,
+            dist_sync_on_step=dist_sync_on_step,
+            process_group=process_group,
+            dist_sync_fn=dist_sync_fn,
+        )
+        if 0 < power < 1:
+            raise ValueError(f"Deviance Score is not defined for power={power}.")
+
+        self.power: float = power
+
+        self.add_state("sum_deviance_score", torch.tensor(0.0), dist_reduce_fx="sum")
+        self.add_state("num_observations", torch.tensor(0), dist_reduce_fx="sum")
+
+    def update(self, preds: Tensor, targets: Tensor) -> None:  # type: ignore
+        """Update metric states with predictions and targets.
+
+        Args:
+            preds: Predicted tensor with shape ``(N,d)``
+            targets: Ground truth tensor with shape ``(N,d)``
```

Comment:
inconsistency is shape docstring (probably also in the functional)
```suggestion
            preds: Predicted tensor with shape ``(N,...)``
            targets: Ground truth tensor with shape ``(N,...)``
```